### DISCUSSION TITLE
A new @session blade directive

### DISCUSSION THREAD

**Discussion by gazmendsahiti** [laravel/framework#45786] [Ideas]
I have seen lots of code in different projects with a repeating `if(session('whatever')) or if(session()->has('whatever'))`, I have been thinking for some time that a @session directive should be shipped out of the box with Laravel. 
If I have the green light I would create the pull request. 

Example: 
`@session('new_user')
       <x-welcome-banner/> 
@endsession
`

**Comment by parallels999** [comment]
Also you could make your own custom directive, example

```php
public function boot()
{
    Blade::if('session', function ($value) {
        return session($value);
    });
}
```
https://laravel.com/docs/9.x/blade#custom-if-statements

**Reply by gazmendsahiti** [reply]
I know

**Reply by parallels999** [reply]
So, just create the pull request, the authors do not read discussions section

**Comment by jrd-lewis** [comment]
This idea was implemented with https://github.com/laravel/framework/pull/49339. 🙂
